### ISSUE TITLE
Counter example doesn't increment the counter.

### ISSUE THREAD

**Issue by zie1ony** [CasperLabs/contract-examples#27]
I have set up the standalone node with `--casper-auto-propose-enabled` and it produces new blocks after every wasm code deploy. I used `counter` example but for some reason I cannot update the produced uref "counter".

These are the steps:

#### Check current block.
```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 show-blocks

------------- block @ 13 ---------------
summary {
  block_hash: "680a90ecae20e333083b0f76564c6d106a3a0b25701172db95496f6570cd90fd"
  ...
```

#### Deploy `counter-define`

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 --port 40401 deploy \
    --from 3030303030303030303030303030303030303030303030303030303030303030 \
    --gas-price 1 \
    --session counterdefine.wasm \
    --payment counterdefine.wasm \
    --nonce 1

Success!
```

#### Wait for block to be produced

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 show-blocks

------------- block @ 14 ---------------
summary {
  block_hash: "55fd3df238c8ffc0b04025921ec4022851471a3e253ea02b2d57170accf504a1"
  header {
    parent_hashes: "680a90ecae20e333083b0f76564c6d106a3a0b25701172db95496f6570cd90fd"
    ...
```

#### Check `count` value. Should be 0.

Get smart contract hash.

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 query-state \
    --block-hash '55fd3df238c8ffc0b04025921ec4022851471a3e253ea02b2d57170accf504a1' \
    --key '3030303030303030303030303030303030303030303030303030303030303030' \
    --type 'address'

account {
  public_key: "3030303030303030303030303030303030303030303030303030303030303030"
  nonce: 0
  known_urefs {
    name: "counter"
    key {
      hash {
        hash: "a4669933ecd6a9a77e2cfaf7b3d6cbe5ef45911905997137ffbcb0c907042a64"
      }
    }
  }
  associated_keys {
    public_key: "3030303030303030303030303030303030303030303030303030303030303030"
    weight: 1
  }
  action_thresholds {
    deployment_threshold: 1
    key_management_threshold: 1
  }
  account_activity {
    key_management_last_used: 0
    deployment_last_used: 0
    inactivity_period_limit: 100
  }
}
```

Get `count` uref address.

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 query-state \ 
    --block-hash '55fd3df238c8ffc0b04025921ec4022851471a3e253ea02b2d57170accf504a1' \
    --key 'a4669933ecd6a9a77e2cfaf7b3d6cbe5ef45911905997137ffbcb0c907042a64' \
    --type 'hash'

contract {
  body:  "..."
  known_urefs {
    name: "count"
    key {
      uref {
        uref: "3a2bee5164b9746b65e50974cdc1466072931d6a9aef3459b5ac3f0588faa513"
        access_rights: READ_ADD_WRITE
      }
    }
  }
  protocol_version: 1
}
```

Get `count` value.

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 query-state \
    --block-hash '55fd3df238c8ffc0b04025921ec4022851471a3e253ea02b2d57170accf504a1' \
    --key '3a2bee5164b9746b65e50974cdc1466072931d6a9aef3459b5ac3f0588faa513' \
    --type 'uref'

int_value: 0
```

`count` value is 0.

#### Deploy `counter-call` 3 times.

This should increment the `count` value.

```bash
ziel@gad ~ $ cat casperlabs-deploy-call.sh 
casperlabs-client --host 127.0.0.1 --port 40401 deploy \
    --from 3030303030303030303030303030303030303030303030303030303030303030 \
    --gas-price 1 \
    --session countercall.wasm \
    --payment countercall.wasm \
    --nonce 1

ziel@gad ~ $ ./casperlabs-deploy-call.sh 
Success!
ziel@gad ~ $ ./casperlabs-deploy-call.sh 
Success!
ziel@gad ~ $ ./casperlabs-deploy-call.sh 
Success!

ziel@gad ~ $ casperlabs-client --host 127.0.0.1 show-blocks

------------- block @ 17 ---------------
summary {
  block_hash: "0fdfb698bd9729adb041897ee8d7252be8e3d5070b285d5d87c1c88dfe88d0b3"

```

#### Check `count` value again.

```bash
ziel@gad ~ $ casperlabs-client --host 127.0.0.1 query-state \
    --block-hash '0fdfb698bd9729adb041897ee8d7252be8e3d5070b285d5d87c1c88dfe88d0b3' \
    --key '3a2bee5164b9746b65e50974cdc1466072931d6a9aef3459b5ac3f0588faa513' \
    --type 'uref'

int_value: 0
```

Value is still 0. What am I doing wrong here?

**Comment by goral09** [comment]
Thank you for your very detailed report and trying out our software! 

A couple of points before we start. It seems like you're using an old version of the node. Can you tell me what version is it? How do I know? You're using `303030...` as a `--from` parameter for the deploys. This was a mocked account that we put in the trie to support deployments before we will create a mechanism for new account creation. 

Let's try to get you updated before we continue debugging this issue. To get the latest version of the node/execution-engine you have to build them by yourself. Instructions how to do it you can find [here](https://github.com/CasperLabs/CasperLabs/).

**Comment by goral09** [comment]
@zie1ony My colleagues pointed out to me that I might be asking too much (building from sources). Maybe this can be fixed easier. Start by first incrementing the `nonce` (1, 2, 3, 4, …) parameter in each subsequent deploy.

**Comment by zie1ony** [comment]
I've just compiled everything from source and now version is `0.4.0 (066311823e0c0f9a0199fe9c3a768a72722ba5d7)`. Previously I've installed from the debian repository. The version was `0.4.0 (51816fb0903e3bbd57c311b162747c5fe168d2f9)`.

Now it works differently.

1. When using `--from 303030...` account nothing happens; there is no longer `known_urefs` section.

2. In `/hack/docker/README.md` `--from` is taken from `account-id-hex` and this is the same as ` validator-id-hex`, but that doesn't get me anywhere as well.
```bash
./casperlabs-client --host 127.0.0.1 query-state --block-hash '9aa...a24' --key 'cca...0c8'
INVALID_ARGUMENT: Value not found: " Account(ccae7167e9f2c48b54485de7fc7543892275e89c085899350d91f880fedbe0c8)"
```
3. Incrementing nonce doesn't help on both versions.

I'll try running from Docker tomorrow.

**Comment by aakoshh** [comment]
@zie1ony thank you for trying it with the build. We're in the process of updating the instructions of starting the node with recent changes. The `303030...` account has been disabled and now one needs to define a genesis account, which is missing from the docs. 

If you built everything with `make docker-build-all` then the `hack/docker` setup will give you an environment you can work with, setting up all the necessary keys. The `ACCOUNT_ID` will be taken from `.casperlabs/genesis/system-account/account-id-hex` which is different then the validator ID, it will be the ID of the account that replaced the `303030...` one.

**Comment by zie1ony** [comment]
I tried to setup few nodes locally in Docker, but couldn't do it as well. I used dev network instead and I can confirm that `counter` example works :)

```console
ziel@gad ~/casperlabs/client/target/universal/stage/bin $ ./casperlabs-client --host deploy.casperlabs.io --port 40401 query-state --key fda0819c1f6d122027338e27482b84803e5fa0573ee5ab1ff9d3c7abf43f6f2e --block-hash 6e68039cf68055c524d5ae0ec6efb4da0e2d63d2a4975717eaf8d802d6fd314a --type uref
int_value: 1
```
I created a new issue https://github.com/CasperLabs/CasperLabs/issues/844 to fix a Docker setup. Closing this one. Thank you all for your time and help.
